eng/common/scripts/Helpers/Metadata-Helpers.ps1

```diff
@@ -80,3 +80,53 @@ function GetPrimaryCodeOwner ([string]$TargetDirectory)
     Write-Warning "No code owner found in $TargetDirectory."
     return $null
 }
+
+function GetDocsMsService($packageInfo, $serviceName) 
+{
+  $service = $serviceName.ToLower().Replace(' ', '').Replace('/', '-')
+  if ($packageInfo.MSDocService) {
+    # Use MSDocService in csv metadata to override the service directory    
+    # TODO: Use taxonomy for service name -- https://github.com/Azure/azure-sdk-tools/issues/1442
+    $service = $packageInfo.MSDocService
+  }
+  Write-Host "The service of package: $service"
+  return $service
+}
+
+function GenerateDocsMsMetadata($language, $langTitle = "", $serviceName, $tenantId, $clientId, $clientSecret, $msService) 
+{
+  if (!$langTitle) {
+    $langTitle = "Azure $serviceName SDK for $language"
+  }
+  $langDescription = "Reference for Azure $serviceName SDK for $language"
+  # Github url for source code: e.g. https://github.com/Azure/azure-sdk-for-js
+  $serviceBaseName = $serviceName.ToLower().Replace(' ', '').Replace('/', '-')
+  $author = GetPrimaryCodeOwner -TargetDirectory "/sdk/$serviceBaseName/"
+  $msauthor = ""
+  if (!$author) {
+    LogError "Cannot fetch the author from CODEOWNER file."
+    return
+  }
+  elseif ($TenantId -and $ClientId -and $ClientSecret) {
+    $msauthor = GetMsAliasFromGithub -TenantId $tenantId -ClientId $clientId -ClientSecret $clientSecret -GithubUser $author
+  }
+  # Default value
+  if (!$msauthor) {
+    LogError "No ms.author found for $author. "
+    $msauthor = $author
```

Comment:
First, it is warning for OPS side. Whenever we have the error here, it is better for us to understand why no author retrieved. If it is temporary issue, then it will be overridden by the next day. If it is a real issue, then we have to look into the API call. So I think make this an acceptable error case help us detect error in an early phase.

Reply:
If OPS won't block us for having an empty `ms.author` then yes, let's keep this as an error so it's visible when it happens in the pipeline but we don't necessarily need to halt execution because of it.